### PR TITLE
Initial Buttons Set up

### PR DESCRIPTION
**Pull request by xnkevinnguyen**
# Description:
Make 3 buttons clickable, current behaviour is not implemented yet.

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Limitations:

Please describe limitations of this PR

# Testing:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code has been formatted with `npm run format` and passes the checks in `npm run check`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by nasadigital** [comment]
@sagarmanchanda What do you think of the current implementation of the Microbit component? I think something similar is used in cpx, as well. To me accessing elements by id seems like it could cause us trouble in the future.
Ideally the elements would be components themselves and we wouldn't access them through ids, but I am not sure how much effort that is.
Not super familiar with these kinds of things, so interested to hear your thoughts!

**Comment by xnkevinnguyen** [comment]
> @sagarmanchanda What do you think of the current implementation of the Microbit component? I think something similar is used in cpx, as well. To me accessing elements by id seems like it could cause us trouble in the future.
> Ideally the elements would be components themselves and we wouldn't access them through ids, but I am not sure how much effort that is.
> Not super familiar with these kinds of things, so interested to hear your thoughts!

@nasadigital  The current implementation is a similar one to the cpx, one potential implementation would be to make components and attach them to the current existing svg or overlay on it. I feel the same way that making regular components would lead to higher code clarity in the future. I could start doing this with LEDS. It also makes sense in the way that I won't have to take care of the rerendering (LEDS)

**Comment by nasadigital** [comment]
> @nasadigital The current implementation is a similar one to the cpx, one potential implementation would be to make components and attach them to the current existing svg or overlay on it. I feel the same way that making regular components would lead to higher code clarity in the future. I could start doing this with LEDS

@xnkevinnguyen Thanks for clarifying! I agree and also it would probably be easier for development. I am not sure if there are any other drawbacks to this approach, though...
For cpx I don't think we should make any changes if everything works.

File: src/view/components/microbit/MicrobitImage.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    buttonElement.onmousedown = e => {
+        eventTriggers.onMouseDown(buttonElement, e);
+    };
+    buttonElement.onmouseup = e => {
+        eventTriggers.onMouseUp(buttonElement, e);
+    };
+
+    buttonElement.onmouseleave = e => {
+        eventTriggers.onMouseLeave(buttonElement, e);
+    };
```
Comment (nasadigital) [inline src/view/components/microbit/MicrobitImage.tsx]: The spacing between the statements is incosistent.

**Review by nasadigital — APPROVED** [review]
LGTM!
shipit

File: src/view/components/cpx/CpxSimulator.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ interface IState {
     cpx: ICpxState;
     play_button: boolean;
 }
-interface IMyProps {
+interface IProps {
```
Comment (sagarmanchanda) [inline src/view/components/cpx/CpxSimulator.tsx]: I don't think you need to define children as a prop, as it already available. So even if you pass empty props here, in your component this.props.children would still be available

File: src/view/components/microbit/MicrobitImage.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    onMouseLeave: (button: HTMLElement, event: Event) => void;
+}
+interface IProps {
+    eventTriggers: EventTriggers;
+}
+
+// Displays the SVG and call necessary svg modification.
+export class MicrobitImage extends React.Component<IProps, any> {
+    componentDidMount() {
+        const svgElement = window.document.getElementById("microbit_svg");
```
Comment (sagarmanchanda) [inline src/view/components/microbit/MicrobitImage.tsx]: Look at how refs and findDOMNode works, there are always preferable ways than getElementById

**Review by sagarmanchanda — CHANGES_REQUESTED** [review]
Can you please check in the first PR and rebase again, I still see a lot of repeated changes here. Thanks! :)

File: src/view/components/cpx/CpxSimulator.tsx (hunk 2 above)
Comment (xnkevinnguyen) [reply src/view/components/cpx/CpxSimulator.tsx]: Yes, i'm making the change to the existing codebase

File: src/view/components/microbit/MicrobitImage.tsx (hunk 3 above)
Comment (xnkevinnguyen) [reply src/view/components/microbit/MicrobitImage.tsx]: I will be using a reference as suggested, it is best practice in case of duplicates

**Pull request closed without merging**